crates/noirc_evaluator/src/ssa_refactor/opt/unrolling.rs

```diff
@@ -0,0 +1,575 @@
+//! This file contains the loop unrolling pass for the new SSA IR.
+//!
+//! This pass is divided into a few steps:
+//! 1. Find all loops in the program (`find_all_loops`)
+//! 2. For each loop:
+//!    a. If the loop is in our list of loops that previously failed to unroll, skip it.
+//!    b. If we have previously modified any of the blocks in the loop,
+//!       restart from step 1 to refresh the context.
+//!    c. If not, try to unroll the loop. If successful, remember the modified
+//!       blocks. If not, remember that the loop failed to unroll and leave it
+//!       unmodified.
+//!
+//! Note that this pass also often creates superfluous jmp instructions in the
+//! program that will need to be removed by a later simplify cfg pass.
+use std::collections::{HashMap, HashSet};
+
+use iter_extended::vecmap;
+
+use crate::ssa_refactor::{
+    ir::{
+        basic_block::BasicBlockId,
+        cfg::ControlFlowGraph,
+        dfg::InsertInstructionResult,
+        dom::DominatorTree,
+        function::Function,
+        instruction::{InstructionId, TerminatorInstruction},
+        post_order::PostOrder,
+        value::ValueId,
+    },
+    ssa_gen::Ssa,
+};
+
+impl Ssa {
+    /// Unroll all loops in each SSA function.
+    /// Panics if any loop cannot be unrolled.
```

Comment:
Is this still the case, re panicking?

Reply:
No, good catch. After the recent commit it should leave them unchanged.